### PR TITLE
sysvar: bypass validation for noop variables

### LINKED ISSUE #31538: show variable command report "ariable 'rpl_semi_sync_master_wait_point' can't be set to the value of ''" when upgrade from 4.0.16 to 5.4.0
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

install tidb cluster with 4.0.16
upgrade to v5.4.0
show variables;

### 2. What did you expect to see? (Required)
command executed successfully
### 3. What did you see instead (Required)
[tidb@101 540]$ mysql -h 172.16.6.77 -P 4000 -uroot -e 'show variables;'
ERROR 1231 (42000) at line 1: Variable 'rpl_semi_sync_slave_enabled' can't be set to the value of ''
[tidb@101 540]$ 
### 4. What is your TiDB version? (Required)

[tidb@101 540]$ mysql -h 172.16.6.77 -P 4000 -uroot -e 'select tidb_version()'
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                       |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v5.4.0
Edition: Community
Git Commit Hash: 2562b030a8d9d5308e3e4b774ec9429255a75c4d
Git Branch: heads/refs/tags/v5.4.0
UTC Build Time: 2022-01-10 11:53:59
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Issue Number: close #31538 

Problem Summary: This is a quick dirty workaround for the issue. It should not change much since affected variables are all no-op. We could fix it more smoothly by versioned sysvars or whatever in the future.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Remove validation for noop variables like `rpl_semi_sync_master_wait_point` in case it is an empty string to workaround upgrading
```

### CODE REVIEW

File: sessionctx/variable/variable.go [hunk 1]
```diff
@@ -267,6 +267,10 @@ func (sv *SysVar) ValidateFromType(vars *SessionVars, value string, scope ScopeF
 	if value == "" && ((sv.AllowEmpty && scope == ScopeSession) || sv.AllowEmptyAll) {
 		return value, nil
 	}
+	// FIXME: hack to bypass some bugs
```
Comment (bb7133) [inline sessionctx/variable/variable.go]:
```suggestion
	// TODO: this is a temporary solution for issue: https://github.com/pingcap/tidb/issues/31538, an elegant solution is needed.
```

**Review by bb7133 — APPROVED** [review]
LGTM

**Comment by xhebox** [comment]
Fixed a test case
/merge

**Comment by xhebox** [comment]
Fix on fix...

/merge

**Pull request merged**